Add Optional Expiry to Auth Tokens (#844)
hashicorp/terraform-provider-tfe: website/docs/r/team_token.html.markdown

* `force_regenerate` - (Optional) If set to `true`, a new token will be
  generated even if a token already exists. This will invalidate the existing
  token!

> * `expired_at` - (Optional) A date and time in which the team token will expire. The expiration date must be passed in
> iso8601 format. If no expiration date is supplied, the expiration date will default to null and never expire.

Reviewer: ```suggestion
* `expired_at` - The token's expiration date. The expiration date must be a date/time string in 
RFC3339 format (For example, "2024-12-31T23:59:59Z"). If no expiration date is supplied, the expiration date will default to null and never expire.
```

## Attributes Reference

* `id` - The ID of the token.
* `token` - The generated token.

> * `expired_at` - The token's expiration date. The expiration date must be passed in
> iso8601 format. If no expiration date is supplied, the expiration date will default to null and never expire.

Reviewer: ```suggestion
```

* `force_regenerate` - (Optional) If set to `true`, a new token will be
  generated even if a token already exists. This will invalidate the existing
  token!
* `expired_at` - (Optional) A date and time in which the team token will expire. The expiration date must be passed in
iso8601 format. If no expiration date is supplied, the expiration date will default to null and never expire.

## Example Usage

Basic usage:

```hcl

> resource "tfe_team" "test" {

Reviewer 2: I think that this is a copy-paste error, this looks like the configuration for a team, not a team token

Author: Great catch! I've updated this.

>   name         = "my-team-name"
>   organization = "my-org-name"

Reviewer 2: ```suggestion
  team_id = "team-id"
```
